### PR TITLE
HttpClient: set CURLOPT_ACCEPT_ENCODING if Accept-Encoding found in request headers

### PR DESCRIPTION
**Pull request by yyuu**
Set `CURLOPT_ACCEPT_ENCODING` if the `Accept-Encoding` header found in the request headers. This is required to enable libcurl's support of `Content-Encoding` (compression of HTTP contents).

### CODE REVIEW

File: extensions/network/HttpClient.cpp [hunk 1]
```diff
@@ -311,6 +311,10 @@ class CURLRaii
             }
         }
 
+        if (request->getAcceptEncoding() != NULL) {
```
Comment (sergey-shambir) [inline extensions/network/HttpClient.cpp]: Please don't call this method twice, save result to local variable. There's also trick `if (const char *acceptEncoding = request->getAcceptEncoding()) {...}`.

Comment (yyuu) [inline extensions/network/HttpClient.cpp]: Thanks for reviewing :) I will update my patch as you suggested.

File: extensions/network/HttpRequest.h [hunk 2]
```diff
@@ -214,7 +215,21 @@ class HttpRequest : public Object
...
+
+    /** Get the value of Accept-Encoding header **/
+    inline const char* getAcceptEncoding() {
+        for (std::vector<std::string>::iterator it = _headers.begin(); it != _headers.end(); ++it) {
+            int colon = it->find_first_of(":");
+            std::string key = it->substr(0, colon-1);
+            std::string val = it->substr(it->find_first_not_of(" \t", colon+1)); // strip leading whitespaces
+            std::transform(key.begin(), key.end(), key.begin(), ::tolower); // case insensitive
+            if(key.compare("accept-encoding") == 0) {
+                return val.c_str();
```
Comment (dumganhar) [inline extensions/network/HttpRequest.h:227]: It's dangerous to return a pointer point to temporary variable(std::string val).

**Comment by cocosQA** [comment]
Dear yyuu:
  We have created a new branch 'v3' to replace branch 'develop', and this PR has been moved from 'develop' branch to 'v3' branch, the new PR is https://github.com/cocos2d/cocos2d-x/pull/6481
The old 'develop' branch will be deleted soon, and this PR will be closed accordingly.

**Pull request closed without merging**